adapters/json_parse.go

```diff
@@ -106,18 +106,23 @@ func getEarlyPath(js *simplejson.Json, path []string) (*simplejson.Json, error)
 }
 
 func arrayGet(js *simplejson.Json, key string) (*simplejson.Json, bool) {
-	input, err := strconv.ParseUint(key, 10, 64)
+	input, err := strconv.ParseInt(key, 10, 64)
 	if err != nil {
 		return js, false
 	}
 	if input > math.MaxInt32 {
```

Comment:
Small suggestion, remove this check and change `ParseInt(key, 10, 64)` to `ParseInt(key, 10, 32)`